File: Arrays/prefix_sum/sub_arr_sum_div_byk.swift
func subarraysDivByK(_ nums: [Int], _ k: Int) -> Int {
        var map: [Int: Int] = [:]
        var sum = 0
        var res = 0
        map[0] = 1
        for i in 0 ..< nums.count {
            sum += nums[i]
            var rem = sum%k
            rem = rem < 0 ? rem + k : rem
            if let count = map[rem] {
                res += count
                map[rem]! += 1
            } else {
                map[rem] = 1
            }
        }
    return res
}

File: Arrays/prefix_sum/sub_arr_sum_div_byk.cpp
#include<iostream>
#include<unordered_map>
#include<vector>

using namespace std;

int subarraysDivByK(vector<int>& nums, int k) {
    unordered_map<int, int> map;
    int sum = 0;
    int res = 0;
    map[0] = 1;

    for(int i: nums) {
        sum += i;
        int rem = sum%k;
        if(rem<0) { rem+=k; }
        if(map[rem] > 0) {
            res+=map[rem];
        }
        map[rem]+=1;
    }
    return res;
}

int main() {
    vector<int> vec = {4,5,0,-2,-3,1};
    int res = subarraysDivByK(vec, 5);
    cout << "RES: " << res << endl; 
}